synology_dsm/synology_dsm.py

```diff
@@ -252,24 +254,29 @@ def _execute_request(self, method, url, params, **kwargs):
                 encoded_params = "&".join(
                     "%s=%s" % (key, quote(str(value))) for key, value in items
                 )
-                resp = self._session.get(
+                response = self._session.get(
                     url, params=encoded_params, timeout=self._timeout, **kwargs
                 )
             elif method == "POST":
-                resp = self._session.post(
+                response = self._session.post(
                     url, params=params, timeout=self._timeout, **kwargs
                 )
 
-            self._debuglog("Request url: " + resp.url)
-            self._debuglog("Request status_code: " + str(resp.status_code))
-            self._debuglog("Request headers: " + str(resp.headers))
+            self._debuglog("Request url: " + response.url)
+            self._debuglog("Request status_code: " + str(response.status_code))
+            self._debuglog("Request headers: " + str(response.headers))
 
-            if resp.status_code == 200:
+            if response.status_code == 200:
                 # We got a DSM response
-                return resp.json()
+                content_type = response.headers.get("Content-Type", "").split(";")[0]
+
+                if content_type in ["application/json", "text/json"]:
+                    return response.json()

```

Comment:
Seems that my DSM gives `Content-Type: text/plain` while the the content is actually json.

Reply:
Seems only some APIs (Camera APIs) give the correct content type. Add "text/plain" to the list might fix the problem.

Reply:
Will do, or maybe you see a better way to handle raw content and json responses ?

Reply:
I think adding `text/plain` is sufficient in this case. Just leave a comment here so we know what's going on in the future.